SNT/NativeMemoryManager.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System.Buffers;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System;
+
+namespace Torch.SNT
+{
+    /// <summary>
+    ///   Class used to wrap the native memory of a Torch Tensor.  
+    /// </summary>
+    internal class NativeMemory<T> : MemoryManager<T>
+    {
+        private int refCount = 0;
+        private IntPtr memory;
```

Comment:
can this be `readonly`?

Reply:
Nope. memory get set to IntPtr.Zero when disposed.